src/GSoft.Extensions.MediatR/RequestLoggingBehavior.cs

```diff
@@ -28,14 +29,38 @@ public async Task<TResponse> Handle(TRequest request, RequestHandlerDelegate<TRe
             var response = await next().ConfigureAwait(false);
 
             watch.Stop();
-            this._logger.RequestSucceeded(requestName, watch.Elapsed.TotalSeconds);
+
+            if (originatingActivity == null)
+            {
+                this._logger.RequestSucceeded(requestName, watch.Elapsed.TotalSeconds);
+            }
+            else
+            {
+                // Make sure the logs being sent are attached to the originating activity
+                originatingActivity.ExecuteAsCurrentActivity(new SuccessfulLoggerState(this._logger, requestName, watch.Elapsed.TotalSeconds), static x =>
+                {
+                    x.Logger.RequestSucceeded(x.RequestName, x.Elapsed);
+                });
+            }
 
             return response;
         }
         catch (Exception ex)
         {
             watch.Stop();
-            this._logger.RequestFailed(ex, requestName, watch.Elapsed.TotalSeconds);
+
+            if (originatingActivity == null)
+            {
+                this._logger.RequestFailed(ex, requestName, watch.Elapsed.TotalSeconds);
+            }
+            else
+            {
+                // Make sure the logs being sent are attached to the originating activity
+                originatingActivity.ExecuteAsCurrentActivity(new FailedLoggerState(this._logger, requestName, watch.Elapsed.TotalSeconds, ex), static x =>
```

Comment:
This was a bug, ending logs were not attached to the same activity of the starting logs. Fixed + changed the unit tests